### PR TITLE
refactor: remove pkg/extensions/search/common and move the code to th…

### PR DESCRIPTION
**Pull request by nicoldr**
…e appropriate packages

**What type of PR is this?**

**Which issue does this PR fix**:
#1342 

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/test/common.go [hunk 1]
```diff
@@ -184,6 +184,14 @@ func MakeHtpasswdFile() string {
 	return MakeHtpasswdFileFromString(content)
 }
 
+// func GetLatestTag(allTags []cvemodel.TagInfo) cvemodel.TagInfo {
```
Comment (andaaron) [inline pkg/test/common.go:187]: Don't forget to clean this up.

File: pkg/test/oci-layout/oci_layout_test.go [hunk 2]
```diff
@@ -431,11 +438,95 @@ func TestExtractImageDetails(t *testing.T) {
...
+		metrics := monitoring.NewMetricsServer(false, log)
+
+		subStore := local.NewImageStore(subRootDir, false,
+			storage.DefaultGCDelay, false, false, log, metrics, nil, nil)
+
+		subStoreMap := make(map[string]storage.ImageStore)
+
+		subStoreMap["/b"] = subStore
+
+		repo, digest := common.GetImageDirAndDigest("image")
```
Comment (andaaron) [inline pkg/test/oci-layout/oci_layout_test.go:528]: This should be moved in the test files of `pkg/common`

File: pkg/test/oci-layout/oci_layout_test.go [hunk 3]
```diff
@@ -431,11 +438,95 @@ func TestExtractImageDetails(t *testing.T) {
...
+		So(len(fixedTags), ShouldEqual, 2)
+
+		fixedTags = cveinfo.GetFixedTags(allTags, append(vulnerableTags, cvemodel.TagInfo{
+			Name:       "taginfo",
+			Descriptor: cvemodel.Descriptor{},
+			Timestamp:  time.Date(2000, time.July, 20, 10, 10, 10, 10, time.UTC),
+		}))
+		So(len(fixedTags), ShouldEqual, 3)
+
+		log := log.NewLogger("debug", "")
```
Comment (andaaron) [inline pkg/test/oci-layout/oci_layout_test.go:511]: Code starting here up until `repo, digest := common.GetImageDirAndDigest("image")` should not be needed anymore. The function it tested was removed.

File: pkg/test/oci-layout/oci_layout_test.go [hunk 4]
```diff
@@ -431,11 +438,95 @@ func TestExtractImageDetails(t *testing.T) {
...
+}
+
+func TestTagsInfo(t *testing.T) {
+	Convey("Test tags info", t, func() {
+		allTags, vulnerableTags := getTags()
+
+		latestTag := ocilayout.GetLatestTag(allTags)
+		So(latestTag.Name, ShouldEqual, "1.0.3")
+
+		fixedTags := cveinfo.GetFixedTags(allTags, vulnerableTags)
```
Comment (andaaron) [inline pkg/test/oci-layout/oci_layout_test.go:501]: Can these tests be moved under `pkg/extensions/search/cve`?

File: pkg/test/oci-layout/oci_layout_test.go [hunk 5]
```diff
@@ -431,11 +438,95 @@ func TestExtractImageDetails(t *testing.T) {
...
+		olu := ocilayout.NewBaseOciLayoutUtils(storeController, testLogger)
 		resDigest, resManifest, resIspecImage, resErr := olu.ExtractImageDetails("zot-test", "latest", testLogger)
 		So(resErr, ShouldEqual, zerr.ErrBlobNotFound)
 		So(string(resDigest), ShouldEqual, "")
 		So(resManifest, ShouldBeNil)
 		So(resIspecImage, ShouldBeNil)
 	})
 }
+
+func getTags() ([]cvemodel.TagInfo, []cvemodel.TagInfo) {
```
Comment (andaaron) [inline pkg/test/oci-layout/oci_layout_test.go:450]: I suggest we move this under `pkg/extensions/search/cve`

File: pkg/test/oci-layout/oci_layout_test.go [hunk 6]
```diff
@@ -431,11 +438,95 @@ func TestExtractImageDetails(t *testing.T) {
...
+	vulnerableTags = append(vulnerableTags, secondTag)
+
+	return tags, vulnerableTags
+}
+
+func TestTagsInfo(t *testing.T) {
+	Convey("Test tags info", t, func() {
+		allTags, vulnerableTags := getTags()
+
+		latestTag := ocilayout.GetLatestTag(allTags)
```
Comment (andaaron) [inline pkg/test/oci-layout/oci_layout_test.go:498]: To test this, I suggest you initialize tag objects in this convey. You only need  the logic in `getTags` up until `vulnerableTags` initialization

**Review by andaaron — APPROVED** [review]
Looks good.

**Comment by andaaron** [comment]
Coverage change is unrelated to this PR (it is dedupe-related).

**Review by andaaron — APPROVED** [review]

**Review by rchincha — APPROVED** [review]
lgtm

**Pull request merged**